Translation support for module "practice"

shprotru opened the pull request:
Next step for #17097

This is initial support for translation for module "practice".

preview:
<details>
<img width="2689" height="1675" alt="image" src="https://github.com/user-attachments/assets/e72631f3-cd59-416e-9e37-a8cb4ba3b250" />
</details>

shprotru committed:
i17097: Initial translation for practice section

shprotru committed:
Update modules.scala

formatting issue

shprotru committed:
i17097: Implemented more translatable symbols

shprotru committed:
i17097: Applying translation

shprotru committed:
i17097: Added section headers translation

shprotru committed:
i17097: fix format issues

shprotru committed:
i17097: translation for section names and section descrions

shprotru committed:
i17097: final fixups

shprotru commented:
@schlawg greetings! Please, give me advices on how can be right translation implemented differ way: where should be stored translation for `sections[].name` and how it should be named?
We create additional translation files `spamdb/data/practice-config.en_US.txt`(contain language code). After that we edit `spamdb/modules/env.py` and `spamdb/modules/study.py` to meke it upload into collection `flag` new documents `practice_en_US`, same name with language code, or directly into document's field `config_en_US`. Or, may be you wish somehow else, just give me instructions how, this is important to not do same work several times) After that we should appeal from code, using currently selected language code to collection or document which contains our translation, if it's absent, than appeal to the default data. Better implement that as stored procedure, that will take an argument from application layer, do all that checks and gettings, and return resulting strings, or implement that on application layer? Additionally we should make possible upload string identifiers to crowdin, and I currently have no ideas how it should be implemented for that repository, and this is an other big question for me.

Generally, I think this should implemented for 3 steps(PR's):
1. Implement necessary stored procedure into `lila-db-seed`, placing(where?) new translation files
2. Change data fetching from `flags` collection, using that procedure in repo `lila`
3. Implement the rest for `lila-db-seed`, like crowdin uploading

Additionally en-GB language code, which you are noticed, is currently seems absent in lila, am I right understand you?

<details>
<img width="935" height="427" alt="Снимок экрана от 2025-08-14 09-08-11" src="https://github.com/user-attachments/assets/9c8d0834-0432-40e5-a317-34d56991c3be" />
</details>

We don't have to be afraid hardcode currently, because I can manually serve all changes for this module and operatively send you PR's on updates, while I will doing implementation based on DB translation mechanisms.

shprotru commented:
@schlawg oh, if I can be somehow helpful, just contact me directly in discord PM or email: `ser3da.p@yandex.ru`. I really want the practice section support translations ❤️

superuser-does commented:
I left a more detailed comment in our internal discussion, and will reiterate it here for visibility.

> any idea when was the last time we edited/added a new practice study? 

The last real edits to the copy were in 2024, with minor adjustments in January 2025. There remains some feedback in #14219 that requires a decision. The decision to effectively freeze the strings should follow some final adjustments to Practice.

We should consider doing a phased release. In a phased approach, we'd first determine which parts of the course we are happy with. They should represent well-written content with an appropriate learning curve. After that, let's make those parts _only_ translateable.

This will be a good impetus to actually complete the course. There have often been ideas about expanding it out, and setting a vision for what the Practice course should be, will give us an idea of when a module can be considered 'final'. For example, we may be looking to make a complete one-volume guide to the game, akin to the _Mammoth Book of Chess_ by Graham Burgess.

In summary, here is what I'd suggest:

1. Set a vision for /practice
2. Determine what pages we are happy with so far (I am making the assumption we will want to revise some parts of the course)
3. Make those we are happy with translateable, with the understanding they will probably never change again.

ornicar committed:
Merge branch 'master' into i17097_practice

ornicar committed:
scala tweaks, use types wherever we can

ornicar committed:
quit pretending that practice sections are dynamic

remove the configuration page and hardcode the entire practice structure

it's inevitable if we translate it with crowdin - I think

shprotru committed:
Merge branch 'master' into i17097_practice

ornicar committed:
Merge branch 'master' into i17097_practice

ornicar committed:
Merge branch 'master' into i17097_practice

shprotru committed:
Update ru-RU.xml

string "Broaden your knowledge"

ornicar committed:
Merge branch 'master' into i17097_practice

ornicar committed:
start integrating practice translation keys into the practice structure

instead of duplicating the structure with translation maps

there's a lot more to do!

ornicar committed:
move more practice translation keys to the practice structure - WIP

doesn't compile but I gotta switch to something else

ornicar committed:
Merge branch 'master' into i17097_practice

ornicar committed:
complete merging practice translation keys into the structure

ornicar committed:
remove a couple of superfluous translation keys

ornicar committed:
remove unnecessary text and translation

ornicar committed:
remove translation/dest/practice/*

these files can only be created and updated by and through crowdin

shprotru committed:
Merge branch 'master' into i17097_practice

schlawg commented:
@superuser-does 
are you happy with the existing strings? thibault gave a link to edit them.

i think they're better than nothing. if they're fine, lichess should either request new changes from shprotru or merge this hard work. old PRs start to smell after a few months.

ornicar committed:
Merge branch 'master' into i17097_practice

shprotru committed:
Merge branch 'master' into i17097_practice

shprotru committed:
Merge branch 'master' into i17097_practice

shprotru committed:
Update i18n.d.ts
